src/Build.UnitTests/BackEnd/TaskBuilder_Tests.cs

```diff
@@ -642,14 +642,14 @@ public void NullMetadataOnOutputItems_InlineTask()
                       </Target>
                     </Project>";
 
-            MockLogger logger = ObjectModelHelpers.BuildProjectExpectSuccess(projectContents, _testOutput);
+            MockLogger logger = ObjectModelHelpers.BuildProjectExpectSuccess(projectContents, _testOutput, LoggerVerbosity.Diagnostic);
             logger.AssertLogContains("[foo: ]");
         }
 
         /// <summary>
         /// If an item being output from a task has null metadata, we shouldn't crash.
         /// </summary>
-        [Fact(Skip = "https://github.com/dotnet/msbuild/issues/6521")]
+        [Fact(Skip = "This test fails when diagnostic logging is available, as deprecated EscapingUtilities.UnescapeAll method cannot handle null value. This is not relevant to non-deprecated version of this method.")]
```

Comment:
This was a interesting regression after reusing loggers from `ProjectCollection`.
`ProjectCollection` creates 2 loggers from those registered: `Microsoft.Build.Evaluation.ProjectCollection.ReusableLogger` based on provided, and `Microsoft.Build.BackEnd.Logging.CentralForwardingLogger` which is `Diagnostic`, and leads to the bug.

Therefore after removing explicit loggers being passed to `Build`, this scenario started to fail too.